### ISSUE TITLE
Subscriptions to start paying for

### ISSUE THREAD

**Issue by kelle** [astropy/project#39]
Once we get our own credit card, what should we buy?!?
- Slack [Update: we now have upgraded Slack via NumFocus.]

Anything else?

**Comment by hamogu** [comment]
Which of those things do we really need? We've done fine without them so far. While some of the things listed here are not  expensive (unlike Slack), I'm worried about the extra work required to keep all those things up-to-date (change user permission, update passwords, update Credit card information, sing-off purchases, classify items for financial reports).

**Comment by hamogu** [comment]
So, I think we should focus on items that save us work such as hosting services or CI pipelines that are less flaky, don't time out as often, don't change their configuration that often etc.

**Comment by kelle** [comment]
If there are things which will make developing easier/faster, I think we should pay for them and try to figure out the admin costs/overheads. We have a team 1Password for keeping track of passwords and credit card info. And we're gonna need an admin for lots of things (see #24), so this would be something they would be responsible for.

**Comment by mhvk** [comment]
From an ethical perspective, I'd feel most compelled to pay for readthedocs and travis (for the former, especially if it comes with not having any ads shown even for those rare few without ad blocking). I'd feel bad about paying any money to microsoft or google; I think as a free software project, we should have some preference towards similarly community spirit driven services. 

Purely from a practical perspective, github itself has become quite slow, not sure if that is partially to push people to paid services, or if one can pay for spiffiness.

**Comment by pllim** [comment]
Is there a place where we can rank/vote these services?

**Comment by kelle** [comment]
@pllim , just let us know your thoughts/votes/priorities in words! If you prefer, please feel free to email  finance@astropy.org.

**Comment by pllim** [comment]
I was just thinking would be nice to have the aggregated rankings public. Individual votes can be anonymous. 🤷‍♀ 

In terms of priorities, here are the ranked reasons from me:

1. Really need that paid feature.
2. Really feel bad we have not been paying (i.e., ReadTheDocs).
3. Would be nice to have some paid features but will survive without them.

I see the point by @mhvk about ethical reasons but if we really can use that paid feature, we can negotiate for Open Source discount/perks.

**Comment by olebole** [comment]
While I think that it is useful to pay companies for their services, I would like to throw in another argument: Astropy is one of the major players nowadays, and the influence we *can* have on the companies is not negleglibe.
If we just pay for a service, then we may be find, but all the smaller project still can't have it. If we try to convince companies that they should open up one or the other property for free (optimally: "free as in speech"), then others would benefit as well.
As an example, Debian is not willing to base their development on top of a non-free service, and so there were many discussions with the Gitlab people to release some formerly propertiary features to the community version.
Obviously, Astropy is in a different situation than Debian, but I would appreciate that before buying in somewhere, we should make an attemtp to free the features we want.

**Comment by Cadair** [comment]
+1 for paying RTD as a community project they could really use the support (and I am sure astropy could use some improved build resources as well, or maybe access to the continuous documentation feature?)

-1 On paying for CI, there are *so many* free services I can't see how you can need to pay for any. (it would be more effective, imo, to pay someone to maintain the CI configs than pay for build time)

**Comment by Cadair** [comment]
Also +1 to supporting free and open source projects over closed solutions, astropy is in a lucky position to have some substantial funding and we should be paying it forward to grow the wider community where we can.

**Comment by pllim** [comment]
Okay, I guess we currently don't really *need* any paid CI features. But we do need a data server that is not GitHub Pages.

**Comment by bsipocz** [comment]
> continuous documentation feature

this is very much what we could use, and was planning to have, but they said no due to out long build time. Given money that limitations I suppose can be lifted.

Otherwise I agree that I can't think of features that we need and can only get by paying for. In the past we had CI limitations, but that's not an issue any longer and fully agree with Stuart that we can cover all our needs with standard, free solutions (that doesn't cost more people time than if we paid for them).

**Comment by Cadair** [comment]
> Okay, I guess we currently don't really _need_ any paid CI features. But we do need a data server that is not GitHub Pages.

As I have said before, I would say S3 is the way to do this, see https://github.com/sunpy/sample-data/ for how we mirror git lfs to S3.

**Comment by bsipocz** [comment]
how do you make sure you use the same region as the CI using that data?

**Comment by pllim** [comment]
Amazon is a big corporation too and they are even worse than other corps when it comes to supporting OSS. Also S3 can catch your wallet off-guard if you are not careful.

**Comment by bsipocz** [comment]
we can also try to reach out to them, as I recall there were some plans for a shared NF proposal/grant with them (maybe it was only brainstormed about)? Maybe @nicole-numfocus knows the details.

**Comment by Cadair** [comment]
> how do you make sure you use the same region as the CI using that data?

We cache it using the CI services cache to reduce bandwidth etc.

**Comment by pllim** [comment]
Also, what about sponsoring Python Software Foundation or something? https://www.python.org/psf/sponsorship/

**Comment by bsipocz** [comment]
> Also, what about sponsoring Python Software Foundation or something? https://www.python.org/psf/sponsorship/

  I feel that's very much beyond our scope.

**Comment by nicole-numfocus** [comment]
> we can also try to reach out to them, as I recall there were some plans for a shared NF proposal/grant with them (maybe it was only brainstormed about)? Maybe @nicole-numfocus knows the details.

@bsipocz if you were referring to NF working with AWS, then yes. They are gifting us some credits to distribute to projects. There will be some info about this going out in the newsletter on Monday so stay tuned. 😄

**Comment by hamogu** [comment]
Trying to summarize:

- Broad agreement to pay RTD
- storage needed, AWS might be a solution (possibly through numfocus)

Not needed:
- CI services (as of now)
- slack (because we have what we need)

**Comment by migueldvb** [comment]
>     * storage needed, AWS might be a solution (possibly through numfocus)

In addition to hosting sample data, it would be useful to build and provide
native astropy packages through AWS instances for distributions where those
packages are not available in the official repositories.

**Comment by hamogu** [comment]
@migueldvb: What distributions are you thinking about? 

My impression is that its typically quite a bit of work to set up, test, debug, and maintain the build scripts for any specific distribution. Unless we find people who are willing to do that and commit to doing it long term, it's not going to happen, even if we have ample AWS instances to run things on.

**Comment by bsipocz** [comment]
Shouldn't that be the job of the distributions? What other projects take up doing packaging themselves outside of the official support?

**Comment by hamogu** [comment]
@bsipocz It depends. For example, for another project, I support macports. I just had a lot of mac users with install problems that all went away once I offered the package through macports. It's also not uncommon for projects to provide their own rpms etc. (e.g. blender offer two windows builds, a linux build and a Mac build, conda has plenty builds for conda, even though conda is also available through some distributions). 

However, that discussion is beyond the scope of this issue. Unless there is a distribution method that we already support now or that we would like to support but that is just held up by not having AWS instances, then it goes here. If we want to support new distributions methods beyond what we do today (pypi, conda-forge), then that should go into a separate issue.

**Comment by mhvk** [comment]
Agreed that this would not seem to be a reason for AWS - even though I'm super-happy with having  most of linux covered by having .deb and .rpm made for us, I think our work should end at making sure `pip install astropy` works for cases where our dependencies can be installed as well (numpy, especially).  Of course, we should not be in the way of anybody packaging for us, as it really helps to catch bugs!

**Comment by bsipocz** [comment]
exactly @mhvk! Others packaging for us is great, but taking up the burden of distributing it beyond the official channels doesn't seem a thing we should start doing (working closely with those packagers is another question, and in fact we already do that with Debian). 

Astropy is big enough that it's part of big distros, including default anaconda, so I don't see strong case for rolling our own on AWS.

**Comment by migueldvb** [comment]
> @migueldvb: What distributions are you thinking about?
> 
> My impression is that its typically quite a bit of work to set up, test, debug, and maintain the build scripts for any specific distribution. Unless we find people who are willing to do that and commit to doing it long term, it's not going to happen, even if we have ample AWS instances to run things on.

Looking at *available* astropy packages in Linux distributions, there are a
number of them that are not included in their respective official repositories.
In my recent experience there are good-quality Arch build scripts available for
astropy and astropy-affiliated packages in community-driven repositories that
allow to compile the packages. These packages will likely not make it to the
official project repositories any time soon so it could be useful to provide
latest versions as pre-built packages, for example for college students using
those platforms. I agree that probably I underestimated the time and effort
required for this task.

**Comment by migueldvb** [comment]
> Shouldn't that be the job of the distributions? What other projects take up doing packaging themselves outside of the official support?

Mozilla builds and distributes several binary versions of firefox [as a nixpkgs overlay](https://github.com/mozilla/nixpkgs-mozilla).  In the python world sublime text hosts [some binary packages for distributions](https://www.sublimetext.com/docs/3/linux_repositories.html).  Some projects maintain build scripts for distribution in a free component of an enterprise product.

**Comment by olebole** [comment]
@migueldvb:
> Looking at _available_ astropy packages in Linux distributions, there are a
> number of them that are not included in their respective official repositories.
> In my recent experience there are good-quality Arch build scripts available for
> astropy and astropy-affiliated packages in community-driven repositories that
> allow to compile the packages. These packages will likely not make it to the
> official project repositories any time soon so it could be useful to provide
> latest versions as pre-built packages, for example for college students using
> those platforms. I agree that probably I underestimated the time and effort
> required for this task.

I can speak here only for Debian (Ubuntu+Mint+Raspbian take their package from there too): The major effort with "official" packages is their continuous maintenance. If someone volunteers to do that for an affiliated package, it is IMO worth to also include their effort into the "official" distribution. At least for Debian, I would welcome and 
support that.

> Mozilla builds and distributes several binary versions of firefox as a nixpkgs overlay. 

Mozilla and Sublime are IMO a bit different from our case since they traditionally don't fit very well into the traditional distributions and require either quite a few patches (Mozilla), or packaging gazillions of dependencies, or are not suitable by their license (sublime).
In contrast, Astropy  really cares about the needs of distributions.

**Comment by bsipocz** [comment]
> for example for college students using those platforms

Astropy currently provides ways for these students, or in fact almost anyone (with the possible exception of folks using super exotic hardware, but I'm sure they then have issue with getting numpy, too) to use the library with two rather standard and generic way, via pip or conda. I still fail to see how the distro packaging is a task/responsibility for the project to maintain and provide infrastructural resources for, rather than task for the distributions.

**Comment by hamogu** [comment]
Can I politely ask to continue the discussion "Should be support more ways of distribution" in a different issue, if any more discussion is needed?
It's not something we want to be paying for right now (as the discussion above has made clear), so it's not part of "subscriptions we want to start paying for" (the title of this issue).

**Comment by eteq** [comment]
A quick note here: it seems like there's pretty universal agreement we should be supporting readthedocs, so the interim finance committee will be working on that ASAP.  Other stuff still needs convergence it seems like.

**Comment by stscicrawford** [comment]
As a note, STScI is currently providing support for cloud storage and distribution of Astropy Data (https://github.com/astropy/astropy-data/issues/91)

**Comment by pllim** [comment]
"is currently providing" might be a little misleading. STScI offered to do it but more things need to happen before it is possible. astropy/astropy-data#91 is still unresolved. 🤷

**Comment by kelle** [comment]
This was a very useful discussion! If there are any specific proposals/requests to buy/upgrade a subscription to something, please open a new issue.

**Comment by teuben** [comment]
> Amazon is a big corporation too and they are even worse than other corps when it comes to supporting OSS. Also S3 can catch your wallet off-guard if you are not careful.

yup, lost $5000 credit once and got overcharged $1500 once. No dead-man-switch
